plugins/utils/harpoon.nix

```diff
@@ -155,27 +158,40 @@ in {
       (
         name: value: {
           term.cmds = value.termCommands;
-          mark.marks = map (mark: {filename = mark;}) value.marks;
+          mark.marks =
+            helpers.ifNonNull' value.marks
+            (
+              map
+              (
+                mark: {
+                  filename = mark;
+                }
+              )
+              value.marks
+            );
         }
       )
       cfg.projects;
 
-    options =
+    setupOptions = with cfg;
       {
         global_settings = {
-          save_on_toggle = cfg.saveOnToggle;
-          save_on_change = cfg.saveOnChange;
-          enter_on_sendcmd = cfg.enterOnSendcmd;
-          tmux_autoclose_windows = cfg.tmuxAutocloseWindows;
-          excluded_filetypes = cfg.excludedFiletypes;
-          mark_branch = cfg.markBranch;
+          save_on_toggle = saveOnToggle;
+          save_on_change = saveOnChange;
+          enter_on_sendcmd = enterOnSendcmd;
+          tmux_autoclose_windows = tmuxAutocloseWindows;
+          excluded_filetypes = excludedFiletypes;
+          mark_branch = markBranch;
         };
 
         inherit projects;
 
-        menu = {
-          inherit (cfg.menu) width height;
-          borderchars = cfg.menu.borderChars;
+        menu = with menu; {
+          inherit
+            width
+            height
+            ;
+          borderchars = borderChars;
```

Comment:
I'd prefer to avoid having nested with foo; with bar; It makes it harder to check where a values comes from, and could easily introduce bugs if we add an option with the same name in `cfg` and in `cfg.menu`

Reply:
You are right (although the innermost scope should prevail... I guess).
Anyway, I removed it.